### PR TITLE
[FuncSpec] Update MinFunctionSize logic

### PR DESCRIPTION
**Pull request by hazzlim**
Always require functions to be larger than MinFunctionSize when SpecializeLiteralConstant is enabled, and increase MinFunctionSize to 500, to prevent excessive triggering of specialisations on small functions.

### CODE REVIEW

**Comment by kiranchandramohan** [comment]
> MinFunctionSize to 400

Nit: It is 500 in the source.

**Comment by hazzlim** [comment]
Reduces the O3 (non-LTO) compile-time of CTMark, measured by the metric of instruction count, when `SpecializeLiteralConstant` is enabled.

```
Benchmark        |  Specs  | Change  |
-----------------+---------+---------+
ClamAV           | 2 -> 2  | -0.01%  |
7zip             | 0 -> 0  |  0.00%  |
kimwitu++        | 0 -> 0  | -0.01%  |
tramp3d-v4       | 0 -> 0  |  0.00%  |
sqlite3          | 0 -> 0  |  0.00%  |
mafft            | 0 -> 0  |  0.00%  |
SPASS            | 1 -> 0  | -0.43%  |
lencod           | 3 -> 0  | -0.39%  |
consumer-typeset | 0 -> 0  | -0.02%  |
Bullet           | 0 -> 0  |  0.00%  |
-----------------+---------+---------+
GEOMEAN          |         | -0.09%  |
```

It has no impact on Release-LTOg builds (either with or without `SpecializeLiteralConstant`), as no specializations are performed.

This change is intended to allow us to enable `SpecializeLiteralConstant` by default, without impacting compile-time.

File: llvm/lib/Transforms/IPO/FunctionSpecialization.cpp [hunk 1]
```diff
@@ -641,12 +641,17 @@ bool FunctionSpecializer::run() {
         Metrics.analyzeBasicBlock(&BB, GetTTI(F), EphValues);
     }
 
+    // When specializing literal constants is enabled, always require functions
+    // to be larger than MinFunctionSize, to prevent excessive specialization.
+    bool RequireMinSize =
```
Comment (momchil-velikov) [inline llvm/lib/Transforms/IPO/FunctionSpecialization.cpp:646]: May add "const" for a little bit of extra self-documentation. Or not.

**Review by momchil-velikov — APPROVED** [review]
LGTM

File: llvm/lib/Transforms/IPO/FunctionSpecialization.cpp (hunk 1 above)
Comment (hazzlim) [reply llvm/lib/Transforms/IPO/FunctionSpecialization.cpp]: Good point - done :)

**Comment by labrinea** [comment]
The test that was added in https://reviews.llvm.org/D135862 will start failing once we enable specilaziation of literal constatnts. You can fix it now or later by adding -funcspec-for-literal-constant=false on the RUN line.

**Comment by hazzlim** [comment]
> The test that was added in https://reviews.llvm.org/D135862 will start failing once we enable specilaziation of literal constatnts. You can fix it now or later by adding -funcspec-for-literal-constant=false on the RUN line.

Good point - I think that it makes sense to fix things / make it explicit in this change. Done :)

**Review by labrinea — APPROVED** [review]

**Comment by labrinea** [comment]
Thanks for the contribution!

**Review by ChuanqiXu9 — APPROVED** [review]

**Pull request merged**